### PR TITLE
Atomic*Array.size method introduced

### LINKED ISSUE #59: No size in atomic arrays
No size getter in `AtomicArray` and `array` field is private:
```
public class AtomicArray<T> internal constructor(size: Int) {
    private val array = Array(size) { atomic<T?>(null) }

    @JsName("get\$atomicfu")
    public operator fun get(index: Int): AtomicRef<T?> = array[index]
}
```

### LINKED ISSUE #149: Provide AtomicArray.length method
Currently, the only method atomic array provides is `get`, making it very hard to e.g. iterate over all its content of the array, get its length and so on, it would be nice to fix that

### PR DESCRIPTION
**Pull request by SokolovaMaria**
Fixes #149
Fixes #59

### CODE REVIEW

File: atomicfu-transformer/src/main/kotlin/kotlinx/atomicfu/transformer/AtomicFUTransformer.kt [hunk 1]
```diff
@@ -466,6 +470,7 @@ class AtomicFUTransformer(
         // Generates static VarHandle field
         private fun vhField(protection: Int, f: FieldInfo) {
             super.visitField(protection or ACC_FINAL or ACC_STATIC, f.fuName, VH_TYPE.descriptor, null, null)
+            super.visitField(protection or ACC_FINAL or ACC_STATIC, f.arrayLengthMethodHandleName, METHOD_HANDLE_TYPE.descriptor, null, null)
```
Comment (elizarov) [inline atomicfu-transformer/src/main/kotlin/kotlinx/atomicfu/transformer/AtomicFUTransformer.kt:473]: I don't really see why it is even needed. We can directly access the underlying array's length without a method handle, cannot we?

File: atomicfu-transformer/src/main/kotlin/kotlinx/atomicfu/transformer/AtomicFUTransformer.kt [hunk 2]
```diff
@@ -718,45 +732,70 @@ class AtomicFUTransformer(
...
                 }
             }
+            if (f.isArray && iv.name == GET_SIZE) {
+                if (!vh) {
+                    // map to j.u.c.a.Atomic*Array length()
+                    iv.owner = descToName(f.fuType.descriptor)
+                    iv.name = "length"
+                } else {
+                    // map to MethodHandles.arrayLength
+                    val getStaticMH = FieldInsnNode(
```
Comment (elizarov) [inline atomicfu-transformer/src/main/kotlin/kotlinx/atomicfu/transformer/AtomicFUTransformer.kt:742]: We can just use `arraylength` bytecode here. There's no need to use a MethodHandle for it.

**Review by elizarov — CHANGES_REQUESTED** [review]

File: atomicfu-transformer/src/main/kotlin/kotlinx/atomicfu/transformer/AtomicFUTransformer.kt (hunk 2 above)
Comment (SokolovaMaria) [reply atomicfu-transformer/src/main/kotlin/kotlinx/atomicfu/transformer/AtomicFUTransformer.kt]: Got it, no need to use MethodHandle for this. I've erased mapping to MethodHandles.arrayLength and generating the static field.

**Review by elizarov — COMMENTED** [review]
Great

**Pull request merged**